src/room_member.cpp

```diff
@@ -230,12 +230,16 @@ void RoomMember::ReceiveLoop() {
                 break;
             }
 
-            peer->DeallocatePacket(packet);
+            if(IsConnected())           //Otherwise they allready got deallocated by the Shutdown
+                peer->DeallocatePacket(packet);
         }
     }
 };
 
 void RoomMember::Join(const std::string& nickname, const std::string& server, uint16_t server_port, uint16_t client_port) {
+    if(receive_thread.get())    // We are still joined
+        Leave();
```

Comment:
I only call here Leave to get recieve_thread to reset(), when we didn't reset it due to a shutdown inside the receive_loop. If we don't reset here, we will get an Segvault later on 
`receive_thread = std::make_unique<std::thread>(&RoomMember::ReceiveLoop, this);`

So this shouldn't be a thing the user has to worry about. Instead of calling Leave(), we could just do all the steps Leave() does and put them here... But i think calling Leave is the best way of doing this